### DISCUSSION TITLE
Multiselect's onChange event is not fired on Select all/Unselect all

### DISCUSSION THREAD

**Discussion by edvard-mit** [primefaces/community#827] [PrimeNG] [answered]
Hi team, with 17.3.0 version p-multiSelect neither onChange neither onSelectAllChange event fired on Select All/ Unselect All action.
[Stackblitz](https://stackblitz.com/edit/pqrhxe?file=src%2Fapp%2Fdemo%2Fmulti-select-basic-demo.html)
`    <p-multiSelect [options]="cities" [(ngModel)]="selectedCities" optionLabel="name" (onChange)="onChanged()" (onSelectAllChange)="onAllChanged()"></p-multiSelect>`

Is it a bug or feature?

**Comment by mehmetcetin01140** [comment, marked as answer]
Hi,

In the example below, clicking select or unselect gets the behavior you expect. 

https://stackblitz.com/edit/pqrhxe-vzq1ad?file=src%2Fapp%2Fdemo%2Fmulti-select-basic-demo.html,src%2Fapp%2Fdemo%2Fmulti-select-basic-demo.ts,src%2Fapp%2Fdemo%2Fmulti-select-basic-demo.scss

**Reply by edvard-mit** [reply]
Hi, 

And is it possible to select/unselect only filtered values, same as default selectAll behaviour works ? 
Thanks in advance.
